nixos/modules/installer/cd-dvd/sd-image-raspberrypi4.nix

```diff
@@ -18,13 +18,20 @@
 
   sdImage = {
     firmwareSize = 128;
+    firmwarePartitionName = "NIXOS_BOOT";
     # This is a hack to avoid replicating config.txt from boot.loader.raspberryPi
     populateFirmwareCommands =
       "${config.system.build.installBootLoader} ${config.system.build.toplevel} -d ./firmware";
     # As the boot process is done entirely in the firmware partition.
     populateRootCommands = "";
   };
 
+  fileSystems."/boot/firmware" = {
+    # This effectively "renames" the loaOf entry set in sd-image.nix
+    mountPoint = "/boot";
```

Comment:
Does all of the rpi firmware need to be in /boot? I was hoping it was just that one .dtb fille.

Reply:
This is not the "FIRMWARE", but the raspberry-pi-boot-chain specific config.txt, kernel, initrd, and old boot entries. And [yes it does](https://github.com/NixOS/nixpkgs/blob/8c521cb815afce585b828f4d72baadb069c88829/nixos/modules/system/boot/loader/raspberrypi/raspberrypi-builder.sh#L14).

Reply:
I'm not sure if I understand. /boot/firmware is set in sd-image.nix, but this overwrites it to mount to `/boot`? Don't we want to avoid putting too much on the vfat filesystem?

Reply:
`sd-image.nix` is built with the assumption that it builds a generic non-raspberry-pi image, which *coincidentally* ends up having the u-boot for the raspberry pi pre-installed. The "FIRMWARE" partition is u-boot for raspberry pi 3. Maybe it needs to be renamed "UBOOT4RPI" :).

Now, the confusing part, I think.

The `sd-image-raspberrypi4` re-uses the `sd-image.nix` stuff. The FAT32 partition here is **not** for u-boot, but rather it is the usual raspberry pi boot chain, with the kernel and initrd on the partition.

What this thing does, is that it it re-uses the definition for the mountpoint `/boot/firmware`, as configured in `sd-image.nix`, and properly mounts the FAT32 partition on `/boot` *as expected with the raspberry pi boot chain* and our raspbperryPi "loader" options.

This is basically cheating and re-using something that wasn't exactly built for use that way.

We want to avoid putting too much in the FAT partition, but it is required with the raspberry pi toolchain.